### PR TITLE
#126F Plugin generalisation & Dropdown plugin

### PR DESCRIPTION
**Pull request by CarlosNZ**
Back-end counterpart to front-end issue #126/PR132: https://github.com/openmsupply/application-manager-web-app/pull/132 

NOTE: Don't merge this until that issue is merged to `master`.

All this does is update the UserRego template to correctly use the new DropdownSelector plugin.

### CODE REVIEW

File: database/insertData.js [hunk 1]
```diff
@@ -172,11 +172,14 @@ const queries = [
...
-                      elementTypePluginCode: "radioChoice"
+                      elementTypePluginCode: "dropdownChoice"
+                      # Change this to "radioChoice" once we've made the plugin
                       category: QUESTION
                       parameters: {
-                        label: "What category of organisation do you wish to join"
+                        label: "Organisation Type"
+                        description: "Select which type of organisation you belong to."
                         options: ["Manufacturer", "Distributor", "Importer"]
+                        validation: { value: true }
```
Comment (nmadruga) [inline database/insertData.js:182]: Why do you need this? I thought this would be the default behaviour

File: database/insertData.js [hunk 2]
```diff
@@ -252,11 +259,9 @@ const queries = [
                       category: QUESTION
                       parameters: {
                         label: "Select Importer"
-                        options: [
-                          "Importer A",
-                          "Importer B",
-                          "Importer C"
-                        ]
+                        placeholder: "Select"
+                        options: ["Importer A", "Importer B", "Importer C"]
```
Comment (nmadruga) [inline database/insertData.js:263]:
```suggestion
                        options: [
                          "Importer A", 
                          "Importer B", 
                          "Importer C"
                        ]
```

I think that's easier to read

**Review by nmadruga — COMMENTED** [review]
Hi, I'm going to test the front-end issue now.

Thanks for fixing this dropdown, I'm very happy to see this one is done now! Just made 2 comments, please have a look

File: database/insertData.js (hunk 2 above)
Comment (CarlosNZ) [reply database/insertData.js:262]: Need to be careful here as the Query editor doesn't want commas when listing vertically, so it's non-standard JSON.

Also -- I don't think we want to get in the habit of over-riding the auto-formatting. I always copy this whole query into the Web GUI to check it's correct, and it auto-formats it -- and we don't want to have to remember to do a bunch of manual changes every time we paste it back.

File: database/insertData.js (hunk 1 above)
Comment (CarlosNZ) [reply database/insertData.js:180]: Yes, it wasn't -- but it is now.

**Review by nmadruga — APPROVED** [review]
Good to go : D

**Pull request merged**